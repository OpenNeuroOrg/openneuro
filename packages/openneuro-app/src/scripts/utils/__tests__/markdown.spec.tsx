import React from "react"
import { render } from "@testing-library/react"
import { Markdown } from "../markdown"

describe("Test <Markdown> component", () => {
  it("safely handles broken HTML tags", () => {
    const brokenTagInput = "* Markdown document\n<br><br>\n * test content"
    const { asFragment } = render(<Markdown>{brokenTagInput}</Markdown>)
    expect(asFragment()).toMatchSnapshot()
  })
  it("filters out disallowed tags", () => {
    const badTags =
      '* Markdown document\n<script type="text/javascript">alert("this should not happen")</script>\n'
    const { asFragment } = render(<Markdown>{badTags}</Markdown>)
    expect(asFragment()).toMatchSnapshot()
  })
  it("allows a href with certain protocols", () => {
    const hrefExample =
      '<a href="https://example.com">Example link that should work.</a>'
    const { asFragment } = render(<Markdown>{hrefExample}</Markdown>)
    expect(asFragment()).toMatchSnapshot()
  })
  it("does not allow href with unknown protocols", () => {
    const hrefExample =
      '<a href="about:memory">Example link that should not work.</a>'
    const { asFragment } = render(<Markdown>{hrefExample}</Markdown>)
    expect(asFragment()).toMatchSnapshot()
  })
  it("filters close-break tags", () => {
    const hrefExample =
      '<br>sample text</br>'
    const { asFragment } = render(<Markdown>{hrefExample}</Markdown>)
    expect(asFragment()).toMatchSnapshot()
  })
})
