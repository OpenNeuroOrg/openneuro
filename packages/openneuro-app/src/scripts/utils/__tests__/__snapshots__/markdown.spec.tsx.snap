// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test <Markdown> component > allows a href with certain protocols 1`] = `
<DocumentFragment>
  <p>
    <a
      href="https://example.com"
    >
      Example link that should work.
    </a>
  </p>
</DocumentFragment>
`;

exports[`Test <Markdown> component > does not allow href with unknown protocols 1`] = `
<DocumentFragment>
  <p>
    <a>
      Example link that should not work.
    </a>
  </p>
</DocumentFragment>
`;

exports[`Test <Markdown> component > filters close-break tags 1`] = `
<DocumentFragment>
  <p>
    <br />
    sample text
    <br />
  </p>
</DocumentFragment>
`;

exports[`Test <Markdown> component > filters out disallowed tags 1`] = `
<DocumentFragment>
  <ul>
    

    <li>
      Markdown document
    </li>
    

  </ul>
</DocumentFragment>
`;

exports[`Test <Markdown> component > safely handles broken HTML tags 1`] = `
<DocumentFragment>
  <ul>
    

    <li>
      Markdown document

      <br />
      <br />
    </li>
    

    <li>
      test content
    </li>
    

  </ul>
</DocumentFragment>
`;
